File: CSharpDefinitions/Samples/PropertyMap.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CSharpDefinitions.Samples.PropertyMap {

    public interface CandleStickSeries {

        SeriesSettings.InnerColor InnerColor { get; }
        SeriesSettings.OpenValueField OpenValueField { get; }
        SeriesSettings.Reduction Reduction { get; }
    }

    public interface BarSeries {

        SeriesSettings.CornerRadius CornerRadius { get; }
        SeriesSettings.MinBarSize MinBarSize { get; }
    }

    public interface StockSeries {

        [DefaultValue("onlyPoint")]
        string HoverMode { get; }

        SeriesSettings.OpenValueField OpenValueField { get; }
        SeriesSettings.Reduction Reduction { get; }
    }

    namespace SeriesSettings {

        [DefaultValue("open")]
        public class OpenValueField : Alias<string> { }

        [DefaultValue(0)]
        public class CornerRadius : Alias<int> { }

        [DefaultValue(0)]
        public class MinBarSize : Alias<int> { }

        public class Reduction : Alias<object> { }

        [DefaultValue("#ffffff")]
        public class InnerColor : Alias<string> { }
    }

}

File: JSDoc/Samples/PropertyMap.js
﻿module.exports = function () {
    /**
     * @class CandleStickSeries
     * @mixes commonProps.innerColor
     * @mixes commonProps.reduction
     * @mixes commonProps.openValueField
     */
    return {
    };
};

module.exports = function () {
    /**
     * @class StockSeries
     * @mixes commonProps.reduction
     * @mixes commonProps.openValueField
     */
    return {

        /**
         * @property
         * @memberof StockSeries
         * @type {string}
         * @default onlyPoint
         */

        hoverMode: "onlyPoint"
    };
};

module.exports = function () {
    /**
     * @class BarSeries
     * @mixes commonProps.cornerRadius
     * @mixes commonProps.minBarSize
     */
    return {
    };
};

/**
 * @name commonProps
 * @mixin
 */
new {

    /**
     * @memberof commonProps
     * @type {int}
     * @default 0
     */
    cornerRadius: 0,

    /**
     * @memberof commonProps
     * @type {int}
     * @default 0
     */
    minBarSize: 0,

    /**
     * @memberof commonProps
     * @type {string}
     * @default #ffffff
     */
    innerColor: "#ffffff",

    /**
     * @memberof commonProps
     * @type {string}
     * @default open
     */
    openValueField: 'open',

    /**
     * @memberof commonProps
     * @type {object}
     */
    reduction: {
        /**
          * @memberof commonProps.reduction
          */
        color: "#000000",

        /**
          * @memberof commonProps.reduction
          */
        level: "close"
    }

}